### ISSUE TITLE
Security Issue - whitelist doesn't remove $populate

### ISSUE THREAD

**Issue by GautierT** [feathersjs-ecosystem/feathers-mongoose#400]
Hi,
There is a big security issue with this package

It seems that `whitelist` doesn't work.

When I construct my model like this : 
```
module.exports = function (app) {
  const options = {
    Model: createModel(app),
    paginate: app.get('paginate')
  }

  // Initialize our service with any options it requires
  app.use('/accounts', new Accounts(options, app))

  // Get our initialized service so that we can register hooks
  const service = app.service('accounts')

  service.hooks(hooks)
}
```

I'm still able to query like this 
```
const query = {
    $populate: "users",
}
await feathersClient.service("accounts").find({ query: query })
```

And users are populated.

**Comment by DaddyWarbucks** [comment]
I don't think the merged PR solves all instances of this issue, unless I am missing something?

See:
https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/2245af73ae3781842d226dac24bf3a22fe8316de/lib/service.js#L127

https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/2245af73ae3781842d226dac24bf3a22fe8316de/lib/service.js#L162

https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/2245af73ae3781842d226dac24bf3a22fe8316de/lib/service.js#L210

**Comment by GautierT** [comment]
@DaddyWarbucks @daffl : https://github.com/feathersjs-ecosystem/feathers-mongoose/pull/402
Can you check this ? 
I did the same as you, for the others methods.

**Comment by J3m5** [comment]
I think $populate simply needs to be removed from the filters [there](https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/2245af73ae3781842d226dac24bf3a22fe8316de/lib/service.js#L18)
These filters are passed down to the [filterQuery method](https://github.com/feathersjs/feathers/blob/a2af38e0eec84c936a420b099aa596e6e98e47c8/packages/adapter-commons/src/service.ts#L142) in the adapter-commons service.
Then these filters are used as a whitelist [there](https://github.com/feathersjs/feathers/blob/a2af38e0eec84c936a420b099aa596e6e98e47c8/packages/adapter-commons/src/filter-query.ts#L83).
Finally, this filterQuery method is called at the beginning of each methods, removing or not the $populate operator.

So basically, what's need to be done in order to fix the issue completely is removing lines 18 to 21.

**Comment by J3m5** [comment]
>  I realized that a lot of this (max/min pagination, stripping out certain query properties etc.) should also be the responsibility of a query validator.

Isn't that already the case with the [getLimit](getLimit) and [cleanQuery](https://github.com/feathersjs/feathers/blob/687e3c7c36904221b2707d0220c0893e3cb1faa9/packages/adapter-commons/src/filter-query.ts#L40) functions for example ?
